src/monopole_gr/monopole_gr.cpp

```diff
@@ -162,29 +163,28 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
   Gradients_t gradients("monopole_gr gradients", NGRAD, npoints);
   Beta_t beta("monopole_gr shift", npoints);
 
-  params.Add("matter", matter);
-  params.Add("matter_h", matter_h);
-  params.Add("matter_cells", matter_cells);
-  params.Add("matter_cells_h", matter_cells_h);
-  params.Add("integration_volumes", integration_volumes);
-  params.Add("integration_volumes_h", integration_vols_h);
-  params.Add("hypersurface", hypersurface);
-  params.Add("hypersurface_h", hypersurface_h);
-  params.Add("lapse_h", alpha_h);
-  // Lapse is mutable because we swap between alpha and alpha_last
-  params.Add("lapse", alpha, true);
-  params.Add("lapse_last", alpha_last, true);
-  // mutable because the reducer is stateful
-  params.Add("matter_reducer", matter_reducer, true);
-  params.Add("volumes_reducer", volumes_reducer, true);
+  params.Add("matter", matter, restart);
+  params.Add("matter_h", matter_h, restart);
+  params.Add("matter_cells", matter_cells, restart);
+  params.Add("matter_cells_h", matter_cells_h, restart);
+  params.Add("integration_volumes", integration_volumes, restart);
+  params.Add("integration_volumes_h", integration_vols_h, restart);
+  params.Add("hypersurface", hypersurface, restart);
+  params.Add("hypersurface_h", hypersurface_h, restart);
+  params.Add("lapse_h", alpha_h, restart);
+  params.Add("lapse", alpha, restart);
+  params.Add("lapse_last", alpha_last, restart);
+  auto mutable_param = parthenon::Params::Mutability::Mutable;
+  params.Add("matter_reducer", matter_reducer, mutable_param);
+  params.Add("volumes_reducer", volumes_reducer, mutable_param);
```

Comment:
And here's all the code setting the monopole solver to have appropriate restart settings